mindsdb/integrations/handlers/confluence_handler/confluence_tables.py

```diff
@@ -0,0 +1,522 @@
+                        f"Unsupported operator '{condition.op}' for column 'page_id'."
+                    )
+
+                condition.applied = True
+
+            if condition.column == "spaceId":
+                if condition.op == FilterOperator.EQUAL:
+                    space_ids = [condition.value]
+
+                elif condition.op == FilterOperator.IN:
+                    space_ids = condition.value
+
+                else:
+                    raise ValueError(
+                        f"Unsupported operator '{condition.op}' for column 'spaceId'."
+                    )
+
+                condition.applied = True
+
+            if condition.column == "status":
+                if condition.op == FilterOperator.EQUAL:
+                    statuses = [condition.value]
+
+                elif condition.op == FilterOperator.IN:
+                    statuses = condition.value
+
+                else:
+                    raise ValueError(
+                        f"Unsupported operator '{condition.op}' for column 'status'."
+                    )
+
+                condition.applied = True
+
+            if condition.column == "title":
+                if condition.op == FilterOperator.EQUAL:
+                    title = condition.value
+
+                else:
+                    raise ValueError(
+                        f"Unsupported operator '{condition.op}' for column 'title'."
+                    )
+
+                condition.applied = True
+
+        pages = client.get_pages(
+            page_ids=page_ids,
+            space_ids=space_ids,
+            statuses=statuses,
+            title=title,
+            limit=limit
+        )
+
+        pages_df = pd.json_normalize(pages, sep="_")
+        pages_df = pages_df[self.get_columns()]
+
+        return pages_df
+
+    def get_columns(self) -> List[str]:
+        """
+        Retrieves the attributes (columns) of the 'chat messages' resource.
+
+        Returns:
+            List[Text]: A list of attributes (columns) of the 'chat messages' resource.
+        """
+        return [
+            "id",
+            "status",
+            "title",
+            "spaceId",
+            "parentId",
+            "parentType",
+            "position",
+            "authorId",
+            "ownerId",
+            "lastOwnerId",
+            "createdAt",
+            "version_createdAt",
+            "version_message",
+            "version_number",
+            "version_minorEdit",
+            "version_authorId",
+            "body_storage_representation",
+            "body_storage_value",
+            "_links_webui",
+            "_links_editui",
+            "_links_tinyui",
+        ]
+
+
+class ConfluenceBlogPostsTable(APIResource):
+    """
+    The table abstraction for the 'blogposts' resource of the Confluence API.
+    """
+    def list(
+        self,
+        conditions: List[FilterCondition] = None,
+        limit: int = None,
+        sort: List[SortColumn] = None,
```

Comment:
Do we need sort and targets?

Reply:
`APIResource` will execute these if they are not handled within the table implementation. Some of the API endpoints allow a single sort column to be passed. Let me see if I can add that in.

Reply:
Sorting via the API enabled (for one column) for the resources that support it (the API endpoints only support a single column). If there are multiple columns in the order by statement, the first is passed to the API and the others will be handled via our internal logic.

I noticed that there might be an issue in our internal logic that messes up the sorted records. I will explore this separately.

Reply:
We can check it in separate PR